### DISCUSSION TITLE
How to connect to a Docker container?

### DISCUSSION THREAD

**Discussion by Nosferican** [community/community#25529] [Actions]
<p>As part of a GitHub action I would like to run a line like</p>
<pre><code>docker run -d -p 4444:4444 --name selenium selenium/standalone-chrome
</code></pre>
<p>and then having it accessible to my application,</p>
<pre><code>response = HTTP.post("$host:4444/wd/hub/session",
                     [("Content-Type" =&gt; "application/json")],
		     JSON3.write(Dict("desiredCapabilities" =&gt; wd.capabilities, wd.kw...)))
</code></pre>
<p>Currently, I get the following error using either</p>
<pre><code>host = "localhost"
</code></pre>
<p>or</p>
<pre><code>host = "host.docker.internal"
</code></pre>
<p>I have been looking at a few related posts (<a href="https://github.community/t/mysql-connection-from-docker-action/17094/5">MySQL connection from Docker action?</a>, <a href="https://github.community/t/docker-container-not-mapping-to-correct-port-connection-refused/15242">Docker container not mapping to correct port - connection refused</a>, <a href="https://github.community/t/communicate-from-service-back-to-job-container/16449">Communicate from service BACK to job container</a>), but haven’t been able to figure out the exact changes to make it work. I have tried with the container name and network alias as well.</p>

**Comment by dakale** [comment, marked as answer]
<p>Hi, I replied in a different thread but will paste my comment here as well:</p>
<p> …</p>
<p>You could run the webdriver as a service container, in which case youd do something like:</p>
<pre><code>services:
  selenium:
    image: selenium/standalone-chrome
</code></pre>
<p> (see <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idservices" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idservices</a>)</p>
<p>That way a docker network will be created, and from inside a container action, you can connect to it via the hostname “selenium” (comes from the service name). So youd just use hostname=selenium, port=4444</p>
<p>Note that in the above example, no port mapping were specified, thats because for user defined bridge networks in docker, all ports are accesible by default, so the application inside the container only needs to actually be listening on 4444</p>
<p>…</p>
<p>We dont currently support adding docker-specific options, like “network”, when running container actions so unfortunately you wouldnt be able to create a network, and run your action attached to that network specifically (something like docker run --network …)</p>
<p>Of course if you are using a javascript action that would be running on the host, so you could connect via localhost (if you ran the selenium container with --network host)</p>
<p>Hope that helps</p>

**Comment by Nosferican** [comment]
<p>Thanks! That helped me better understand the service <a href="https://github.com/actions/example-services" rel="nofollow noopener">example-services</a>.</p>
<p>I had some issues with the VM way, but the container one worked great. Here is my final workflow file.</p>
<pre><code>name: CI

on: push

jobs:
  ci:
    container: node:current
    services:
      selenium:
        image: selenium/standalone-chrome
        options: --health-cmd '/opt/bin/check-grid.sh'
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        julia-version: [1.3.0]
        julia-arch: [x64]
        os: [ubuntu-latest]
    steps:
      - uses: actions/checkout@v1.0.0
      - uses: julia-actions/setup-julia@latest
        with:
          version: ${{ matrix.julia-version }}
      - uses: julia-actions/julia-buildpkg@latest
      - uses: julia-actions/julia-runtest@latest
      - uses: julia-actions/julia-uploadcodecov@latest
        env:
          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
      - uses: julia-actions/julia-docdeploy@releases/v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
</code></pre>
